### PR TITLE
Make `PythonComplex` a subclass of `PyccelInternalFunction`

### LINKED ISSUE #1703: Fixing class inheritance of PythonComplex
## Description
`PythonComplex` represents a call to Python's native `complex()` function. This class accepts arguments to construct complex number objects. To accurately reflect that this is a function call and will be displayed as a function in C/Fortran, it should be represented as a `PyccelInternalFunction`. The previous code doesn't result in any errors but i believe it is a good enhancement.

### Class Prototype Before:
```python
class PythonComplex(TypedAstNode):
```

### Class Prototype After:
```python
class PythonComplex(PyccelInternalFunction):
```

### PR DESCRIPTION
**Pull request by Farouk-Echaref**
Change the base class of `PythonComplex` from `TypedAstNode` to `PyccelInternalFunction`. This fixes #1703.

### CODE REVIEW

**Review by EmilyBourne — APPROVED** [review]

File: AUTHORS [hunk 1]
```diff
@@ -28,3 +28,4 @@ Contributors
 * Soukaina Hikma
 * Mudit Loya
 * Priyabrata Mondal
+* Farouk ECH-CHAREF
```
Comment (yguclu) [inline AUTHORS:31]: We do not capitalize last names.

**Review by yguclu — CHANGES_REQUESTED** [review]

File: AUTHORS (hunk 1 above)
Comment (yguclu) [reply AUTHORS:31]: Sorry, I meant that we do not use ALL CAPS for names.

Comment (yguclu) [reply AUTHORS:31]: Would this work for you?
```suggestion
* Farouk Ech-Charef
```

Comment (Farouk-Echaref) [reply AUTHORS:31]: Yes it's fine, thanks for the clarification

Comment (Farouk-Echaref) [reply AUTHORS:31]: Should I commit the suggested change, or is it for you to decide?

Comment (yguclu) [reply AUTHORS:31]: You can commit the suggested change

**Review by yguclu — APPROVED** [review]

**Pull request merged**